pkg/farm/list_builder.go

```diff
@@ -70,21 +58,36 @@ func (l *listLocal) build(ctx context.Context, images map[entities.BuildReport]e
 	refs := []string{}
 	for image, engine := range images {
 		image, engine := image, engine
-		pullOptions := entities.PullToLocalOptions{
-			ImageID:     image.ID,
-			SaveFormat:  image.SaveFormat,
-			Destination: l.localEngine,
+		pushOptions := entities.PushToRegistryOptions{
+			ImageID:       image.ID,
+			ManifestName:  l.listName,
+			Authfile:      l.options.authfile,
+			SkipTLSVerify: l.options.skipTLSVerify,
 		}
 		pullGroup.Go(func() error {
-			logrus.Infof("copying image %s", image.ID)
-			defer logrus.Infof("copied image %s", image.ID)
-			ref, err := engine.PullToLocal(ctx, pullOptions)
+			logrus.Infof("pushing image %s", image.ID)
+			defer logrus.Infof("pushed image %s", image.ID)
+			var finalErr error
+			// If the push fails for some reason, try the push again
+			// 5 times before finally failing
+			ref, err := engine.PushToRegistry(ctx, pushOptions)
 			if err != nil {
-				return fmt.Errorf("pulling image %q to local storage: %w", image, err)
+				count := 0
+				for err != nil && count < 5 {
+					ref, err = engine.PushToRegistry(ctx, pushOptions)
+					if err != nil {
+						fmt.Printf("error pushing image %s, retrying...\n", image.ID)
+					}
+					count++
+					finalErr = err
+				}
```

Comment:
Doesn't the eventual call to `.Push()` implement retry logic of its own?

Reply:
I don't think so, without this retry logic it was failing after one attempt. I added this retry to hopefully bypass any random network errors that could occur while trying to push, may not be the best way to do it but something to avoid failing after one attempt if something weird/unrelated happens during the push.

Reply:
`pkg/api/handlers/libpod.PushImage()` and `pkg/api/handlers/compat.PushImage()` both call `pkg/domain/infra/abi.ImageEngine.Push()`, which calls `github.com/containers/common/libimage.Runtime.Push()`, which calls `github.com/containers/common/libimage.copier.copy()`, which uses `github.com/containers/common/pkg/retry.IfNecessary()`, so there's some retrying wired in for certain errors.  In any case we don't want to be adding new custom retry logic if we can avoid it.